### PR TITLE
Enable copy on write by default

### PR DESCRIPTION
**Pull request by fjetter**
Closes https://github.com/dask-contrib/dask-expr/issues/918

~I'm not sure if we need the pinning but https://github.com/dask-contrib/dask-expr/issues/918#issuecomment-1973414969 let me think that this is a safe choice s.t. we don't need any compat code.~

### CODE REVIEW

**Comment by fjetter** [comment]
@phofl I am getting an error that concerns me a little


`test_apply_or_transform_shuffle_multilevel[<lambda>1]`

```
E           ValueError: The columns in the computed data do not match the columns in the provided metadata
E             Extra:   []
E             Missing: ['x', 'y']

```

**Comment by phofl** [comment]
This is a legit bug, but pandas does what it is supposed to do. I don't understand where we are wrong yet

**Comment by phofl** [comment]
Yikes this is a mean bug on our end

**Comment by phofl** [comment]
https://github.com/dask-contrib/dask-expr/pull/933 should fix all related issues

File: pyproject.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ requires-python = ">=3.9"
 dependencies = [
     "dask == 2024.2.1",
     "pyarrow",
-    "pandas >= 2",
+    "pandas >= 2.1",
```
[the remark this answers is not in the document]
Comment (phofl) [reply pyproject.toml:31]:
```suggestion
    "pandas >= 2",
```

[the remark this answers is not in the document]
Comment (phofl) [reply pyproject.toml:31]: I double checked the release notes, there shouldn't be anything in there that hurts us, 2.1 only contains a handful of CoW related changes that were mostly further optimisations

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**